dashboard/src/components/ErrorAlert/ErrorSelector.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from "react";
+import { definedNamespaces } from "../../shared/Namespace";
+import {
+  ConflictError,
+  ForbiddenError,
+  IRBACRole,
+  NotFoundError,
+  UnprocessableEntity,
+} from "../../shared/types";
+import { PermissionsErrorAlert, UnexpectedErrorAlert } from "./";
+import { namespaceText } from "./helpers";
+
+interface IErrorSelectorProps {
+  error: Error;
+  // Resource causing the failure (e.g. "Application my-wordpress")
+  resource: string;
+  // Action related to the error (e.g. create, delete)
+  action?: string;
+  // Default permissions that the component should have
+  // returned in case it's a ForbiddenError and the message
+  // doesn't contain specific RBAC roles
+  defaultRequiredRBACRoles?: {
+    [action: string]: IRBACRole[];
+  };
+  // Namespace in case the error is namespaced
+  namespace?: string;
+}
+
+const ErrorSelector: React.SFC<IErrorSelectorProps> = props => {
+  const { namespace, error, defaultRequiredRBACRoles, action, resource } = props;
+  switch (error.constructor) {
+    case ConflictError:
```

Comment:
I am not convinced about the need of having this error type. It seems too specific when it could come under the umbrella of  `UnprocessableEntity`. Actually that's the error (422) that the server should return since it is a data validation error. 

I personally think that allowing the client to call ErrorSelector with a `UnprocessableEntity` error and a custom title would work best. 

In any case, if you think that it is worth maintaining this error type, go ahead.

Reply:
I understand, but what I am arguing for is to not to treat a "release name invalid or taken" as something different than a validation error 422. That way we do not need to maintain a different use case for such a specific scenario.

Looking at the [409 error description](https://httpstatuses.com/409) to me, it does not seem to fit this purpose either, since it seems to be mainly focused on resource updates.

That said, it seems that the return code is set in Tiller-proxy so it might not make sense to make any change in that regard and carry on with the Conflict error handling.

Reply:
This is what I meant with the 422 vs 409 error https://groups.google.com/d/msg/api-craft/fQWIs3_UwJE/rUqyqWaxbCkJ

Reply:
oh, I didn't know about that discussion. In this case (tiller-proxy) I mimic K8s behavior. They use a 409 when the resource already exists:

```
curl --data $JSON -H "Content-Type: application/json" -X POST http://127.0.0.1:8001/api/v1/namespaces/default/pods/
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {

  },
  "status": "Failure",
  "message": "pods \"nginx\" already exists",
  "reason": "AlreadyExists",
  "details": {
    "name": "nginx",
    "kind": "pods"
  },
  "code": 409
}%
```